hooks/pre_commit/05-puppet_certs_exist.rb

```diff
@@ -0,0 +1,27 @@
+def error(message)
+  say message
+  logger.error message
+  kafo.class.exit 101
+end
+
+def param_value(mod, name)
+  param(mod, name).value if param(mod, name)
+end
+
+server_crl_path = param_value('foreman', 'server_ssl_crl')
+puppet_server_implementation = param_value('puppet', 'server_implementation')
+puppet_passenger = param_value('puppet', 'server_passenger')
+cert_dir = param_value('puppet', 'server_ssl_dir')
+puppet_ca_cert = File.join(cert_dir, 'ca/ca_crt.pem')
+puppet_enabled = kafo.module('puppet').enabled?
+
+client_message = "- is Puppet already installed without Puppet CA? You can remove the existing certificates with 'rm -rf #{cert_dir}' to get Puppet CA properly configured."
+
+if cert_dir && File.directory?(cert_dir)
```

Comment:
I don't think this is precise enough, as any time the agent has run, you might have the directory (since it'll create a CSR).  This will particularly be the case on a Debian installation where the package install usually starts the Puppet agent service.